packages/sdk-auth/src/auth.js

```diff
@@ -53,12 +52,24 @@ export default class SdkAuth {
     this.INTROSPECT_URI = '/oauth/introspect'
   }
 
-  static _getFetcher(configFetch: ?ConfigFetch): ConfigFetch {
-    if (!configFetch && typeof fetch === 'undefined')
+  static _getFetcher(fetcher: ?typeof fetch): typeof fetch {
+    if (!fetcher && typeof fetch === 'undefined')
       throw new Error(
         '`fetch` is not available. Please pass in `fetch` as an option or have it globally available.'
       )
-    return configFetch || fetch
+    let fetchFunction: typeof fetch
+    if (fetcher) {
+      fetchFunction = fetcher
+    } else {
+      // `fetcher` is set here rather than the destructuring to ensure fetch is
+      // declared before referencing it otherwise it would cause a `ReferenceError`.
+      // For reference of this pattern: https://github.com/apollographql/apollo-link/blob/498b413a5b5199b0758ce898b3bb55451f57a2fa/packages/apollo-link-http/src/httpLink.ts#L49
+
+      /* eslint-disable */
+      fetchFunction = fetch
+      /* eslint-enable */
```

Comment:
do we disable here because of the reassignment? Doesn't flow allow casting, maybe?